main.go

```diff
@@ -98,7 +98,14 @@ func main() {
 		os.Exit(1)
 	}
 
-	//controllersLog := rootLog.Named("controllers")
+	controllersLog := rootLog.Named("controllers")
+	if err = (&controllers.DeploymentValidator{
+		Client: mgr.GetClient(),
+		Log:    controllersLog.Named("deployment-validator").Sugar(),
+	}).SetupWebhookWithManager(mgr); err != nil {
+		setupLog.Error("unable to create webhook", zap.NamedError("stack", err))
+		os.Exit(1)
+	}
 	//+kubebuilder:scaffold:builder
```

Comment:
I wrote this wiring manually but it's ~compatible with kubebuilder style wiring.